test/abstractarray.jl

```diff
@@ -1280,3 +1280,25 @@ end
     @test Int[0 t...; t... 0] == [0 1 2; 1 2 0]
     @test_throws ArgumentError Int[t...; 3 4 5]
 end
+
+@testset "issue #39896, modified getindex " begin
+    for arr = ([1:10;], reshape([1.0:16.0;],4,4), reshape(['a':'h';],2,2,2))
+        for inds = (2:5, Base.OneTo(5), BigInt(3):BigInt(5), UInt(4):UInt(3))
+            @test arr[inds] == arr[collect(inds)]
+            @test arr[inds] isa AbstractVector{eltype(arr)}
+        end
+    end
+    
+    for arr = ([1], reshape([1.0],1,1), reshape(['a'],1,1,1))
+        @test arr[true:true] == [arr[1]]
+        @test arr[true:true] isa AbstractVector{eltype(arr)}
+        @test arr[false:false] == []
+        @test arr[false:false] isa AbstractVector{eltype(arr)}
+    end
+    
```

Comment:
This line shouldn’t contain whitespace.

Reply:
Corrected that. Should I squash the previous commits for indentation into one? @sostock @jishnub

Reply:
I think all the commits will be squashed while merging anyway, so it shouldn't be necessary

Reply:
No need to do that, this PR needs to be squashed anyway to be merged.

Reply:
Okay then:)